Add frontend for JWT refresh tokens

stveit opened the pull request:
Adds a frontend for managing JWT refresh tokens. You can see which tokens exist, expire existing ones and generate new ones.

Created new issue for audit log #3405, this PR is big enough

The code duplication warning is legit, but its duplicate from the old token code which will be deleted when JWTs are fully implemented and ready to take over for the old token system.

Also adds permission related fields to the JWTRefreshToken model, since these will be used when creating a token. We store the relevant permissions there so they can be displayed as token metadata. They should never be changed (hence why you cant update them in the Edit form) because the values in the actual token are the ones that matter, and those will not be updated when the database is updated. If you want new permissions, you must make a new token.

stveit commented on tests/integration/jwt_refresh_endpoint_test.py:
```diff
@@ -0,0 +1,185 @@
+@pytest.fixture(scope="module")
+def private_key() -> str:
+    """Yields a private key in PEM format"""
+    key = """-----BEGIN PRIVATE KEY-----
```
Dummy key. But a more relevant point is that there are already tests that define PEM keys, so instead of redefining keys here its probably better to restructure the fixtures to allow this test file access to those fixtures

lunkwill42 reviewed (changes requested):
This seems to work reasonably well, with a caveat (as discussed on Slack):

The refresh endpoint and the useradmin panel seems to read `jwt.conf` on-the-fly, so any changes take effect without reloading the web server.  *However*, the auth middleware is setup at process startup (since it's configured via `settings.py`), and remains statically configured throughout the process' lifetime.

This gave me the following problem on my first test:
1. I started the web server, and browsed the useradmin panel, which told me I needed to configure `jwt.conf`.
2. I configured `jwt.conf` and created the keys.
3. I was able to issue a new refresh token through the UI.
4. I was able to issue a new access token from the API.
5. However, the API would deny all attempts to access it using the access token, claiming that my token's `iss` claim was invalid.

Reloading the web server seems to get everything in sync.

Other than that, attempting to access an endpoint that is not part of my claim, I get this error:

> Authentication credentials were not provided

This is sort-of incorrect: I did *provide* credentials, they just were just the wrong ones.

I'll try to do some more extensive testing and add more comments later...

stveit commented:
> Other than that, attempting to access an endpoint that is not part of my claim, I get this error:
> 
>     Authentication credentials were not provided
> 
> This is sort-of incorrect: I did provide credentials, they just were just the wrong ones.

I imagine this is caused by the `JWTPermission` class that handles permissions, so it should probably have its own issue

stveit commented:
Made changes to how inactive/revoked is shown: I first added one info box for if a token is active and one for if it is revoked, but it was pretty confusing if you had a token that was technically active (current time between `nbf` and `exp` claim), but was also revoked. As an end user I dont think it would be clear how a token can be both revoked and active at the same time.

So I ended up combining Revoked into the Status info. Now it will either show `active`, `inactive` or `revoked`, with revoked being red. I made the same change to the token list, removing the Revoked column entirely and integrating it into the Status column.

stveit commented:
The token model didnt actually have a `created` field yet, so I added that and included this data in the details view and token list view.

I also fixed a bunch of other smaller things, like which columns should hide if the screen gets smaller etc.
